Move block and IRs memory pool to rv

qwe661234 opened the pull request:
Because the block map isn't used in JIT mode, we need to move the block and IRs memory pool to rv structure.

jserv reviewed (changes requested):
Refine the Git commit messages, eliminating any grammar mistakes.

RinHizakura commented on src/riscv.c:
```diff
@@ -20,19 +20,14 @@ static void block_map_init(block_map_t *map, const uint8_t bits)
-    assert(map);
-    for (uint32_t i = 0; i < map->block_capacity; i++) {
-        block_t *block = map->map[i];
+    block_map_t map = rv->block_map;
```
I think this should be a direct reference(`block_map_t *map = &rv->block_map`) instead of having another instance on the stack?

qwe661234 committed:
Move the block and IRs memory pool to rv sturcture

Because the block map isn't used in JIT mode, we need to move the block
and IRs memory pool to rv structure.
